### PR TITLE
Adds compile task

### PR DESCRIPTION
**Pull request by ThadHouse**
Reenables travis builds, just without linking.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -240,6 +240,15 @@ model {
                 }
             }
         }
+        compileCpp(Task) {
+            $.binaries.each { binary ->
```
Comment (JLLeitschuh) [inline build.gradle:244]: What's the `$` mean in groovy?

**Comment by ThadHouse** [comment]
It uses other sources of the Gradle model setup as an input. So this gives you access to all the model binaries.

File: build.gradle (hunk 1 above)
Comment (333fred) [reply build.gradle:244]: `$` is a gradle-specific symbol, it means model root.

Comment (JLLeitschuh) [reply build.gradle:244]: Thanks!

**Pull request closed without merging**